src/Database/LSMTree/Model/Normal/Admin.hs

```diff
@@ -0,0 +1,383 @@
+
+--
+-- API
+--
+
+lookups ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint k
+     , Model.SomeSerialisationConstraint v
+     )
+  => [k]
+  -> TableHandle k v blob
+  -> m [Model.LookupResult k v (Model.BlobRef blob)]
+lookups ks th = do
+    table <- guardTableHandleIsClosed th
+    pure $ Model.lookups ks table
+
+type RangeLookupResult k v blobref = Model.RangeLookupResult k v blobref
+
+rangeLookup ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint k
+     , Model.SomeSerialisationConstraint v
+     )
+  => Model.Range k
+  -> TableHandle k v blob
+  -> m [RangeLookupResult k v (Model.BlobRef blob)]
+rangeLookup r th = do
+    table <- guardTableHandleIsClosed th
+    pure $ Model.rangeLookup r table
+
+updates ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint k
+     , Model.SomeSerialisationConstraint v
+     , Model.SomeSerialisationConstraint blob
+     , C k v blob
+     )
+  => [(k, Model.Update v blob)]
+  -> TableHandle k v blob
+  -> m ()
+updates ups th@TableHandle{..} = do
+  table <- guardTableHandleIsClosed th
+  let table' = Model.updates ups table
+  modify (\m -> m {
+      tableHandles = Map.insert tableHandleID (SomeTable table') (tableHandles m)
+    })
+
+inserts ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint k
+     , Model.SomeSerialisationConstraint v
+     , Model.SomeSerialisationConstraint blob
+     , C k v blob
+     )
+  => [(k, v, Maybe blob)]
+  -> TableHandle k v blob
+  -> m ()
+inserts = updates . fmap (\(k, v, blob) -> (k, Model.Insert v blob))
+
+deletes ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint k
+     , Model.SomeSerialisationConstraint v
+     , Model.SomeSerialisationConstraint blob
+     , C k v blob
+     )
+  => [k]
+  -> TableHandle k v blob
+  -> m ()
+deletes = updates . fmap (,Model.Delete)
+
+retrieveBlobs ::
+     ( MonadState Model m
+     , MonadError Err m
+     , Model.SomeSerialisationConstraint blob
+     )
+  => TableHandle k v blob
+  -> [Model.BlobRef blob]
+  -> m [blob]
+retrieveBlobs th refs = do
+    table <- guardTableHandleIsClosed th
+    pure $ Model.retrieveBlobs table refs
+
+{-------------------------------------------------------------------------------
+  Snapshots
+-------------------------------------------------------------------------------}
+
+data Snapshot = Snapshot SUT.TableConfig SomeTable
+  deriving Show
+
+--
+-- API
+--
+
+snapshot ::
+     ( MonadState Model m
+     , MonadError Err m
+     , C k v blob
+     )
+  => SUT.SnapshotName
+  -> TableHandle k v blob
+  -> m ()
+snapshot name th@TableHandle{..} = do
+    table <- guardTableHandleIsClosed th
+    snaps <- gets snapshots
+    when (Map.member name snaps) $
+      throwError (ErrSnapshotExists tableHandleID name)
```

Comment:
That's a reasonable semantics, but we'll have to support deleting snapshots :-)

And as Oleg said, listing them too would be sensible.